File: backend/schemas/places.py
"""Pydantic schemas for Google Places API."""
from pydantic import BaseModel
from typing import List, Optional


class AutocompleteSuggestion(BaseModel):
    """Single autocomplete suggestion."""
    description: str
    place_id: str


class AutocompleteResponse(BaseModel):
    """Response for autocomplete endpoint."""
    suggestions: List[AutocompleteSuggestion]


class PlaceDetailsResponse(BaseModel):
    """Response for place details endpoint."""
    place_id: str
    name: str
    address: str
    lat: float
    lng: float

File: mobile/src/api/places.ts
import apiClient from './client';

export interface AutocompleteSuggestion {
  description: string;
  place_id: string;
}

export interface AutocompleteResponse {
  suggestions: AutocompleteSuggestion[];
}

export interface PlaceDetails {
  place_id: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
}

export const placesApi = {
  autocomplete: async (query: string, city?: string): Promise<AutocompleteSuggestion[]> => {
    const params = new URLSearchParams({ q: query });
    if (city) {
      params.append('city', city);
    }
    const response = await apiClient.get<AutocompleteResponse>(`/places/autocomplete?${params.toString()}`);
    return response.data.suggestions;
  },

  getDetails: async (placeId: string): Promise<PlaceDetails> => {
    const response = await apiClient.get<PlaceDetails>(`/places/details?place_id=${placeId}`);
    return response.data;
  },
};

